src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -319,6 +320,17 @@ def structure_sheaf : sheaf CommRing (Spec.Top R) :=
       (sheaf_condition_equiv_of_iso (structure_presheaf_comp_forget R).symm
         (structure_sheaf_in_Type R).sheaf_condition), }
 
+@[simp] lemma res_apply (U V : opens (Spec.Top R)) (i : V ⟶ U) (s) (x : V) :
+  ((structure_sheaf R).presheaf.map i.op s).1 x = (s.1 (i x) : _) :=
+rfl
+
+lemma germ_ext {U V : opens (Spec.Top R)} {x : Spec.Top R} {hxU hxV}
+  (W : opens (Spec.Top R)) (hxW : x ∈ W) (iWU : W ⟶ U) (iWV : W ⟶ V) {sU sV}
+  (ih : (structure_sheaf R).presheaf.map iWU.op sU = (structure_sheaf R).presheaf.map iWV.op sV) :
+  (structure_sheaf R).presheaf.germ ⟨x, hxU⟩ sU = (structure_sheaf R).presheaf.germ ⟨x, hxV⟩ sV :=
```

Comment:
This isn't a problem that's newly introduced here, but I wish we had some mechanism by which we could get away with writing `(structure_sheaf R).map ...` instead of `(structure_sheaf R).presheaf.map ...` (and similarly for `germ`).

Reply:
Also, this lemma has nothing to do with the structure sheaf, and should just be proved right after `presheaf.germ`.

Possibly we should even tag it as `ext`...

Reply:
This requires a concrete category actually, and then everything becomes messier.

Reply:
It seems straightforward: I did this as https://github.com/leanprover-community/mathlib/pull/4047/commits/8f0f64d601cf5e43c4b40d5d22bd86e89c996a5d